From 51766abe8bb8dd9851f31e120ff57640f76d95ee Mon Sep 17 00:00:00 2001
From: Amisha Jain <quic_amisjain@quicinc.com>
Date: Thu, 23 Jan 2025 12:51:22 +0530
Subject: [PATCH] UPSTREAM:obex: Update the MAP Client version to 1.4 in SDP
 record

This change is required in below PTS testcase:
1. MAP/MCE/SGSIT/ATTR/BV-10-C
Attribute GSIT - Bluetooth Profile Descriptor List, MAP 1.4

Current MAP Client version 1.2 is being deprecated and withdrawn
from BT Sig, so it is mandatory to update the version.

No additional changes are needed for supporting the new version.

Upstream-Status: Backport https://git.kernel.org/pub/scm/bluetooth/bluez.git/commit/?id=1b37809b1df771fcfc2c46a8ece765359615ad28
Signed-off-by: Amisha Jain <quic_amisjain@quicinc.com>

---
 doc/supported-features.txt | 3 ++-
 src/profile.c              | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/doc/supported-features.txt b/doc/supported-features.txt
index 8b71bfe..5318dc6 100644
--- a/doc/supported-features.txt
+++ b/doc/supported-features.txt
@@ -40,7 +40,8 @@ FTP			1.3		Client, Server
 OPP			1.2		Client, Server
 SYNCH			1.1		Client
 PBAP			1.1		Client, Server
-MAP			1.0		Client, Server
+MAP			1.0		Server
+MAP			1.4		Client
 
 HID			1.1		Host
 
diff --git a/src/profile.c b/src/profile.c
index 4855630..e84df1e 100644
--- a/src/profile.c
+++ b/src/profile.c
@@ -2161,7 +2161,7 @@ static struct default_settings {
 		.mode		= BT_IO_MODE_ERTM,
 		.authorize	= true,
 		.get_record	= get_mns_record,
-		.version	= 0x0102
+		.version	= 0x0104
 	},
 };
 
-- 
2.34.1

